### PR TITLE
gitlab-housekeeping if MR status is unchcked for automerge - check it harder

### PR DESCRIPTION
**Pull request by maorfr**
when listing MRs, some come back with `merge_status: unchcked`. calling the merge request endpoint gets this information.

follows up on #2673

ref: https://docs.gitlab.com/ee/api/merge_requests.html#merge-requests-list-response-notes

### CODE REVIEW

**Comment by maorfr** [comment]
this MR was closed by a local run with these changes: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/45183

File: reconcile/gitlab_housekeeping.py [hunk 1]
```diff
@@ -119,8 +119,12 @@ def handle_stale_items(dry_run, gl, days_interval, enable_closing, item_type):
     for item in items:
         item_iid = item.attributes.get("iid")
         item_labels = item.attributes.get("labels")
-        if AUTO_MERGE in item_labels and item.merge_status == MRStatus.CANNOT_BE_MERGED:
-            close_item(dry_run, gl, enable_closing, item_type, item)
+        if AUTO_MERGE in item_labels:
+            if item.merge_status == MRStatus.UNCHECKED:
+                # this call triggers a status recheck
+                item = gl.get_merge_request(item_iid)
+            if item.merge_status == MRStatus.CANNOT_BE_MERGED:
```
Comment (janboll) [inline reconcile/gitlab_housekeeping.py:126]: nit: IMHO `elif` would make it a little easier to read

Comment (maorfr) [reply reconcile/gitlab_housekeeping.py:126]: we are getting a new item to process, so an `elif` would not be logically correct?

**Review by janboll — APPROVED** [review]

File: reconcile/gitlab_housekeeping.py (hunk 1 above)
Comment (janboll) [reply reconcile/gitlab_housekeeping.py:126]: Whoops. Never would have expected that and thus overlooked it.

**Pull request merged**